Client/Frontend/Home/ActivityStreamTopSitesCell.swift

```diff
@@ -248,15 +248,17 @@ class ASHorizontalScrollCell: UITableViewCell {
 
     override func layoutSubviews() {
         super.layoutSubviews()
-        let layout = collectionView.collectionViewLayout as! HorizontalFlowLayout
+        reloadData().uponQueue(DispatchQueue.main) { _ in
```

Comment:
Doing this here will reload the view much more than is required. Every time you switch pages it reloads the table. This creates more animations than it should.